rustls/src/msgs/handshake.rs

```diff
@@ -1352,12 +1363,17 @@ impl Codec<'_> for CertificateEntry {
 
     fn read(r: &mut Reader) -> Result<Self, InvalidMessage> {
         Ok(Self {
-            cert: CertificateDer::read(r)?,
+            cert: certificate_der_into_owned(CertificateDer::read(r)?),
             exts: Vec::read(r)?,
         })
     }
 }
 
+// FIXME(perf) sub-optimal in the owned case; switch to `CertificateDer::into_owned` when available
+fn certificate_der_into_owned(cert: CertificateDer<'_>) -> CertificateDer<'static> {
+    CertificateDer::from(cert.to_vec())
+}
```

Comment:
I think this FIXME can be resolved, I see [CertificateDer::into_owned](https://docs.rs/rustls-pki-types/latest/rustls_pki_types/struct.CertificateDer.html#method.into_owned) landed.